generate_docs.sh

```diff
@@ -23,7 +23,7 @@ install_prefix="$source_dir/install"
 
 if [ "$skip_checks" = false ]; then
     # Clean-up build & install directory
-    if [ -e $source_dir/build/ ] || [ -e $install_prefix ] ; then
+    if [ -d $source_dir/build ] && [ "$(ls $source_dir/build/default)" ] || [ -d $install_prefix ] ; then
```

Comment:
Why not `-d` instead of `ls` ?

Reply:
-d tests whether directory exists; `ls` will check whether directory is empty. `make clean` doesn't delete `build/default`; So we need `ls` for that.

Reply:
Hm? But `make clean` does delete `build/default`.

Reply:
No :) [make distclean](https://github.com/dronecore/DroneCore/blob/develop/Makefile#L162) deletes it; not [make clean](https://github.com/dronecore/DroneCore/blob/develop/Makefile#L166).
```
user@ubuntu:~/karma/fork/test/DroneCore [develop] $ ls build/default/
CMakeCache.txt  cmake_install.cmake  CTestTestfile.cmake         dronecore-unittests  gmock.pc       gtest.pc  integration_tests  libs      plugins            stop_px4_sitl.sh
CMakeFiles      core                 dronecore-integrationtests  gmock_main.pc        gtest_main.pc  include   libdronecore.so    Makefile  start_px4_sitl.sh
user@ubuntu:~/karma/fork/test/DroneCore [develop] $ make clean
user@ubuntu:~/karma/fork/test/DroneCore [develop] $ ls build/default/
user@ubuntu:~/karma/fork/test/DroneCore [develop] $ 
user@ubuntu:~/karma/fork/test/DroneCore [develop] $ make distclean
user@ubuntu:~/karma/fork/test/DroneCore [develop] $ ls build/default/
ls: cannot access 'build/default/': No such file or directory
```

Reply:
@julianoes let `make clean` do `rm -rf build` ?

Reply:
Oh, I didn't realize there was a `distclean`. How does this work? I didn't realize we changed this.

Reply:
Yeah, it just deletes `build` & `logs` directories, if any.
```
distclean:
	@rm -rf build/
	@rm -rf logs/
```